homeassistant/components/mqtt/__init__.py

```diff
@@ -401,13 +401,17 @@ def _mqtt_on_subscribe(self, _mqttc, _userdata, mid, granted_qos):
 
     def _mqtt_on_message(self, _mqttc, _userdata, msg):
         """Message received callback."""
-        _LOGGER.debug("received message on %s: %s",
-                      msg.topic, msg.payload.decode('utf-8'))
-        self.hass.bus.fire(EVENT_MQTT_MESSAGE_RECEIVED, {
-            ATTR_TOPIC: msg.topic,
-            ATTR_QOS: msg.qos,
-            ATTR_PAYLOAD: msg.payload.decode('utf-8'),
-        })
+        try:
+            _LOGGER.debug("received message on %s: %s",
+                          msg.topic, msg.payload.decode('utf-8'))
```

Comment:
It would make sense to do the decoding only once and wrap that in a `try…except` block.